js/reference.js

```diff
@@ -2127,25 +2151,103 @@
 
                     // check url length limit if not first one;
                     if (i != 0 && new Reference(module.parse(baseUri + currentPath + (i != 0 ? ';' : '') + filter), self.table.schema.catalog).contextualize.compact._getReadPath().value.length > module.URL_PATH_LENGTH_LIMIT) {
-                        referencePaths.push(currentPath);
+                        // any more filters will go over the url length limit so save the current path and count
+                        // then clear both to start creating a new path
+                        referencePathObjs.push({
```

Comment:
If our goal is to keep the rows that didn't get deleted in the list of "selectedRows" in chaise, you would need to also keep the array of original tuples and not just the count (or maybe an array of their indexes and then create the tuples based on that)

Reply:
I decided to keep an array of objects with the column name as one value and the key value as the other. This lets me look up the row based on key column information